packit/constants.py

```diff
@@ -3,13 +3,18 @@
 
 DG_PR_COMMENT_KEY_SG_PR = "Source-git pull request ID"
 DG_PR_COMMENT_KEY_SG_COMMIT = "Source-git commit"
+
+# we store downstream content in source-git in this subdir
+DISTRO_DIR = ".distro"
+SG_CONFIG_NAME = "source-git.yaml"
 CONFIG_FILE_NAMES = [
     ".packit.yaml",
     ".packit.yml",
     ".packit.json",
     "packit.yaml",
     "packit.yml",
     "packit.json",
+    f"{DISTRO_DIR}/{SG_CONFIG_NAME}",
```

Comment:
> I would rather prefere having this set for `packit source-git` sub-commands only, so that this path is not searched when running other Packit commands.

Actually, it's useful also with `packit srpm`, where you also need to add the `-c/--config`.

> Doing the above would also help avoid confusion when using source-git commands on a repo which also has Packit configured.

It's "searched" only as a last resort when none of the above is found so there should not be any confusion (i.e. it would not be "searched" at all) on a repo that uses one of the above configs.

Reply:
What will happen if a project uses Packit to build and test their code upstream (that is they have `packit.yaml`) *and* they also have a source-git repo in Fedora? Will `packit source-git update-dist-git` do the correct thing without explicitly specifying which config (`packit.yaml` or `.distro/source-git.yaml`) to use?

Reply:
No, it won't do the "correct thing" because it will take `packit.yaml`.
But that's how it works now as well, so no change in this use-case, AFAICT.

So, how is it supposed to work if there are both, `.distro/source-git.yaml` & `packit.yaml` (from upstream).
IMHO, in such case the `.distro/source-git.yaml` should be preferred since the repo is a source-git repo and the `packit.yaml` is the upstream's one.
So, should I actually put the `.distro/source-git.yaml` as a first thing on the `CONFIG_FILE_NAMES` list?

Reply:
> So, should I actually put the `.distro/source-git.yaml` as a first thing on the `CONFIG_FILE_NAMES` list?

Yes, this would be a more useful change. But as I said above, I think it would make more sense and help avoid future trouble if defaults for `packit source-git` would be kept separate.

Reply:
> But as I said above, I think it would make more sense and help avoid future trouble if defaults for `packit source-git` would be kept separate.

And I said that it's useful also for `packit srpm`, but I can leave `packit srpm` as is since all I wanted to achieve with this PR is to make things easier for hardly.
Anyway, I don't see a way how to easily make this change only for `packit source-git` since it uses common package config discovery functions.
I'd need to change [get_local_package_config](https://github.com/packit/packit/blob/8df4f423df97b6163e54c6213395a9c5b7d75253/packit/config/package_config.py#L271) and [find_packit_yaml](https://github.com/packit/packit/blob/8df4f423df97b6163e54c6213395a9c5b7d75253/packit/config/package_config.py#L217) so they prefer source-git config if used via `packit source-git`.
I'm willing to do that if you tell me what future trouble you foresee.
Otherwise, please unblock the review since I've moved the `.distro/source-git.yaml` to the top of `CONFIG_FILE_NAMES`. Thank you.